x-pack/plugins/transform/public/app/sections/transform_management/components/transform_list/action_delete.tsx

```diff
@@ -71,17 +85,96 @@ export const DeleteAction: FC<DeleteActionProps> = ({ items, forceDisable }) =>
+          values={{ count: items.length }}
+        />
+      </p>
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem>
+          {
+            <EuiSwitch
+              data-test-subj="transformBulkDeleteIndexSwitch"
+              style={{ paddingBottom: 10 }}
+              label={i18n.translate(
+                'xpack.transform.actionDeleteTransform.bulkDeleteDestinationIndexTitle',
+                {
+                  defaultMessage: 'Delete destination indexes',
+                }
+              )}
+              checked={deleteDestIndex}
+              onChange={toggleDeleteIndex}
+            />
+          }
+        </EuiFlexItem>
+        <EuiFlexItem>
+          {
+            <EuiSwitch
+              data-test-subj="transformBulkDeleteIndexPatternSwitch"
+              label={i18n.translate(
+                'xpack.transform.actionDeleteTransform.bulkDeleteDestIndexPatternTitle',
+                {
+                  defaultMessage: 'Delete destination index patterns',
+                }
+              )}
+              checked={deleteIndexPattern}
+              onChange={toggleDeleteIndexPattern}
+            />
+          }
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </>
+  );
+
+  const deleteModalContent = (
+    <>
+      <p>
+        <FormattedMessage
+          id="xpack.transform.transformList.deleteModalBody"
+          defaultMessage="Are you sure you want to delete this transform?"
+        />
+      </p>
+      <EuiFlexGroup direction="column" gutterSize="none">
+        <EuiFlexItem>
+          {userCanDeleteIndex && (
+            <EuiSwitch
+              data-test-subj="transformDeleteIndexSwitch"
+              style={{ paddingBottom: 10 }}
+              label={i18n.translate(
+                'xpack.transform.actionDeleteTransform.deleteDestinationIndexTitle',
+                {
+                  defaultMessage: 'Delete destination index {destinationIndex}',
+                  values: { destinationIndex: items[0] && items[0].config.dest.index },
+                }
+              )}
+              checked={deleteDestIndex}
+              onChange={toggleDeleteIndex}
+            />
+          )}
+        </EuiFlexItem>
+        <EuiFlexItem>
+          {userCanDeleteIndex && indexPatternExists && (
+            <EuiSwitch
+              data-test-subj="transformDeleteIndexPatternSwitch"
+              label={i18n.translate(
+                'xpack.transform.actionDeleteTransform.deleteDestIndexPatternTitle',
+                {
+                  defaultMessage: 'Delete index pattern {destinationIndex}',
+                  values: { destinationIndex: items[0] && items[0].config.dest.index },
+                }
+              )}
+              checked={deleteIndexPattern}
+              onChange={toggleDeleteIndexPattern}
+            />
+          )}
+        </EuiFlexItem>
+      </EuiFlexGroup>
+    </>
```

Comment:
this block looks the same expect i18n strings are different. I'd prefer to have only one and just have conditional rendering of the labels.